### PR TITLE
fix: aria-selected should not be set on role link

### PR DESCRIPTION
**Pull request by vvscode**
Covers https://github.com/react-navigation/react-navigation/issues/9418

### CODE REVIEW

**Comment by vvscode** [comment]
@SampsonCrowley FYI

**Comment by Elter71** [comment]
Hi, thanks for your contributes. add same changes to packages/bottom-tabs/src/views/BottomTabItem.tsx, on master

File: packages/drawer/src/__tests__/views/DrawerItem.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  );
+
+  expect(getByRole('button').props.accessibilityState).toBeDefined();
+  expect(getByRole('button').props.accessibilityState).toEqual({
+    selected: true,
+  });
+});
+
+it('passes accessibilityState={{ selected: false }} to PlatformPressable from non-focused DrawerItem', () => {
+  Platform.OS = 'android';
```
Comment (Elter71) [inline packages/drawer/src/__tests__/views/DrawerItem.test.tsx:31]: I think it is not needed. It's working fine without defining OS

**Review by Elter71 — CHANGES_REQUESTED** [review]

**Comment by satya164** [comment]
Hello, the PR got closed automatically due to some changes in the main branch. Would you be able to send a new PR? Thank you.

**Pull request closed without merging**